lib/src/configuration.dart

```diff
@@ -0,0 +1,85 @@
+// Copyright 2019 Google LLC. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import 'package:source_span/source_span.dart';
+
+import 'ast/node.dart';
+import 'ast/sass.dart';
+import 'util/limited_map_view.dart';
+import 'util/unprefixed_map_view.dart';
+import 'value.dart';
+
+/// A set of variables meant to configure a module by overriding its
+/// `!default` declarations.
+///
+/// There are two types of configuration: an explicit one that's created by
+/// the `with` clause of a `@use` rule and an implicit one that's created any
+/// time a file containing a `@forward` rule is imported.
+///
+/// Both types of configuration pass through `@forward` rules, but explicit
+/// configurations will cause an error if attempting to use them on a module
+/// that has already been loaded, while implicit configurations will be silently
+/// ignored in this case.
+class Configuration {
+  /// A map from variable names (without `$`) to values.
+  ///
+  /// When this is empty, it may be unmodifiable, so [Map.remove] should not be
+  /// called on this.
+  final Map<String, ConfiguredValue> values;
+
+  /// Whether or not this configuration is implicit.
+  final bool isImplicit;
+
+  Configuration(this.values, {this.isImplicit = false});
+
+  /// The empty configuration, which indicates that the module has not been
+  /// configured.
+  const Configuration.empty()
+      : values = const {},
+        isImplicit = false;
+
+  bool get isEmpty => values.isEmpty;
+  bool get isNotEmpty => values.isNotEmpty;
```

Comment:
Typically classes that aren't implementing core library interfaces don't define `isNotEmpty`. (Defining `isNotEmpty` at all was a design mistake—having a different way to negate exactly one boolean getter doesn't make sense—but there's no going back on that for the core libraries at this point.)